### ISSUE TITLE
bug?: unable to migrate manifest v2 => v3

### ISSUE THREAD

**Issue by rxliuli** [mozilla/addons-linter#4943]
### Describe the problem and steps to reproduce it:

(Please include as many details as possible.)

### What happened?

I previously wrote a firefox extension that uses manifest v2, and now I want to upgrade to manifest v3, after finishing everything, I get an error after uploading

<img width="1032" alt="image" src="https://github.com/mozilla/addons-linter/assets/24560368/493d2dff-718a-4a5a-88c7-d0deb3894227">

But when I look at the full validation report, it doesn't show any errors

<img width="1011" alt="image" src="https://github.com/mozilla/addons-linter/assets/24560368/9b795cab-e81a-4857-b741-84e3cfcf70a8">

### What did you expect to happen?

Ability to migrate v2 to v3 and publish

### Anything else we should know?

(Please include screenshots and any relevant files.)

**Comment by Rob--W** [comment]
1. Which extension are you trying to update?
2. What is the content of your manifest.json file?

**Comment by rxliuli** [comment]
1. https://addons.mozilla.org/firefox/addon/joplin-search-integration/
2.

```json
{
  "name": "Joplin Search Integration",
  "version": "0.5.3",
  "description": "When using search, related Joplin notes are also displayed in the search results.",
  "manifest_version": 3,
  "background": {
    "scripts": [
      "service-worker-loader.js"
    ],
    "type": "module"
  },
  "content_scripts": [
    {
      "js": [
        "assets/main.ts-loader-4c4fd344.js"
      ],
      "matches": [
        "<all_urls>"
      ]
    }
  ],
  "host_permissions": [
    "http://localhost:27583/*",
    "http://localhost:41184/*"
  ],
  "permissions": [
    "storage"
  ],
  "icons": {
    "16": "src/assets/16.png",
    "32": "src/assets/32.png",
    "48": "src/assets/48.png",
    "128": "src/assets/128.png"
  },
  "options_ui": {
    "page": "src/options/index.html"
  },
  "web_accessible_resources": [
    {
      "matches": [
        "<all_urls>"
      ],
      "resources": [
        "assets/index-a11a3338.js",
        "assets/loadConfig-0f0a73d9.js",
        "assets/browser-polyfill-716a599d.js",
        "assets/main.ts-bb8ebee4.js"
      ],
      "use_dynamic_url": true
    }
  ],
  "browser_specific_settings": {
    "gecko": {
      "id": "joplin-search-integration@rxliuli.com",
      "strict_min_version": "109.0"
    }
  }
}
```

**Comment by Rob--W** [comment]
The extension ID of your add-on is `{77acd154-9528-47d8-b340-dbd61729064d}`

You should put that in your manifest file instead of `joplin-search-integration@rxliuli.com` 
. I believe that the error message also states this issue with your extension.

**Comment by rxliuli** [comment]
> The extension ID of your add-on is `{77acd154-9528-47d8-b340-dbd61729064d}`
> 
> You should put that in your manifest file instead of `joplin-search-integration@rxliuli.com` . I believe that the error message also states this issue with your extension.

Yes, but using this id will cause another error

<img width="990" alt="image" src="https://github.com/mozilla/addons-linter/assets/24560368/dc6f52e0-fdc9-4ef1-b01f-e8d38e8aa9a1">

**Comment by Rob--W** [comment]
You dropped the `{` and `}`. These are part of your extension ID and should be included.
